contracts/exchange/EndemicPrivateSale.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+import "./EndemicExchangeCore.sol";
+
+error PrivateSaleExpired();
+error InvalidSignature();
+error InvalidPrivateSale();
+error PriceNotMatchWithProvidedEther();
+
+contract EndemicPrivateSale is
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    EndemicExchangeCore
+{
+    using AddressUpgradeable for address;
+
+    bytes32 private constant PRIVATE_SALE_TYPEHASH =
+        keccak256(
+            // solhint-disable-next-line max-line-length
+            "PrivateSale(address nftContract,uint256 tokenId,address seller,address buyer,uint256 price,uint256 deadline)"
+        );
+
+    bytes32 private constant EIP712_DOMAIN_TYPEHASH =
+        keccak256(
+            "EIP712Domain(string name,string version,uint256 chainId,address verifyingContract,bytes32 salt)"
+        );
+
+    bytes32 private constant SALT_HASH = keccak256("Endemic Exchange Salt");
+
+    string private constant DOMAIN_NAME = "Endemic Exchange";
+
+    bytes32 public DOMAIN_SEPARATOR;
+
+    // Maps nftContract -> tokenId -> seller -> buyer -> invalidated.
+    mapping(address => mapping(uint256 => mapping(address => mapping(address => bool))))
+        private privateSaleInvalidated;
+
+    event PrivateSaleFinalized(
+        address indexed nftContract,
+        address indexed seller,
+        address buyer,
+        uint256 indexed tokenId,
+        uint256 price,
+        uint256 totalFees
+    );
+
+    function __EndemicPrivateSale___init_unchained() internal {
+        _updateConfiguration();
+    }
+
+    function _updateConfiguration() internal {
+        DOMAIN_SEPARATOR = keccak256(
+            abi.encode(
+                EIP712_DOMAIN_TYPEHASH,
+                keccak256(bytes(DOMAIN_NAME)),
+                keccak256(bytes("1")),
+                block.chainid,
+                address(this),
+                SALT_HASH
+            )
+        );
+    }
```

Comment:
Do we need extra function here? Can we put this login directly inside of __EndemicPrivateSale___init_unchained if we are not reusing it?

Reply:
Of course, we don't reuse it I just put it separately for sake of better explanatory what we are doing when the initial method is called